File: 2022/01/07/main.go
package main

import "math/rand"

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
type Solution struct {
	l []int
}

func Constructor(head *ListNode) Solution {
	t := []int{}
	for head != nil {
		t = append(t, head.Val)
		head = head.Next
	}
	return Solution{
		l: t,
	}
}

func (this *Solution) GetRandom() int {
	r := rand.Intn(len(this.l))
	return this.l[r]
}

/**
 * Your Solution object will be instantiated and called as such:
 * obj := Constructor(head);
 * param_1 := obj.GetRandom();
 */

File: 2022/01/07/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:

    def __init__(self, head: Optional[ListNode]):
        self.l = []
        while head:
            self.l.append(head.val)
            head = head.next

    def getRandom(self) -> int:
        r = int(random.random() * len(self.l))
        return self.l[r]


# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()